packages/enzyme-test-suite/test/RSTTraversal-spec.jsx

```diff
@@ -97,71 +68,59 @@ describe('RSTTraversal', () => {
-      expect(nodeHasProperty(<div foo={0} />, 'foo', '0')).to.equal(true);
-      expect(nodeHasProperty(<div foo={-0} />, 'foo', '-0')).to.equal(true);
-      expect(nodeHasProperty(<div foo={1} />, 'foo', '0')).to.equal(false);
-      expect(nodeHasProperty(<div foo={2} />, 'foo', '-0')).to.equal(false);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', 0)).to.equal(true);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', +0)).to.equal(true);
+      expect(nodeHasProperty(<div foo={-0} />, 'foo', -0)).to.equal(true);
+      expect(nodeHasProperty(<div foo={1} />, 'foo', 0)).to.equal(false);
+      expect(nodeHasProperty(<div foo={2} />, 'foo', -0)).to.equal(false);
     });
 
     it('should work with empty strings', () => {
       expect(nodeHasProperty(<div foo={''} />, 'foo', '')).to.equal(true);
-      expect(nodeHasProperty(<div foo={''} />, 'foo', '""')).to.equal(true);
       expect(nodeHasProperty(<div foo={'bar'} />, 'foo', '')).to.equal(false);
-      expect(nodeHasProperty(<div foo={'bar'} />, 'foo', '""')).to.equal(false);
     });
 
     it('should work with NaN', () => {
-      expect(nodeHasProperty(<div foo={NaN} />, 'foo', 'NaN')).to.equal(true);
-      expect(nodeHasProperty(<div foo={0} />, 'foo', 'NaN')).to.equal(false);
+      expect(nodeHasProperty(<div foo={NaN} />, 'foo', NaN)).to.equal(true);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', NaN)).to.equal(false);
     });
 
     it('should work with null', () => {
-      expect(nodeHasProperty(<div foo={null} />, 'foo', 'null')).to.equal(true);
-      expect(nodeHasProperty(<div foo={0} />, 'foo', 'null')).to.equal(false);
+      expect(nodeHasProperty(<div foo={null} />, 'foo', null)).to.equal(true);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', null)).to.equal(false);
     });
 
     it('should work with false', () => {
-      expect(nodeHasProperty(<div foo={false} />, 'foo', 'false')).to.equal(true);
-      expect(nodeHasProperty(<div foo={0} />, 'foo', 'false')).to.equal(false);
+      expect(nodeHasProperty(<div foo={false} />, 'foo', false)).to.equal(true);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', false)).to.equal(false);
     });
 
     it('should work with ±Infinity', () => {
-      expect(nodeHasProperty(<div foo={Infinity} />, 'foo', 'Infinity')).to.equal(true);
-      expect(nodeHasProperty(<div foo={0} />, 'foo', 'Infinity')).to.equal(false);
-      expect(nodeHasProperty(<div foo={-Infinity} />, 'foo', '-Infinity')).to.equal(true);
-      expect(nodeHasProperty(<div foo={0} />, 'foo', '-Infinity')).to.equal(false);
-    });
-
-    it('should throw when un unquoted string is passed in', () => {
-      const node = $(<div title="foo" />);
-
-      expect(() => nodeHasProperty(node, 'title', 'foo')).to.throw();
+      expect(nodeHasProperty(<div foo={Infinity} />, 'foo', Infinity)).to.equal(true);
+      expect(nodeHasProperty(<div foo={Infinity} />, 'foo', +Infinity)).to.equal(true);
```

Comment:
let's add a case where `Infinity` fails to match a prop value of `Infinity`?

Reply:
Done.

Reply:
It'd also be good to ensure NaN and Infinity don't equate, since both are not finite